### PR TITLE
[Bug 1233] -Fix Made Changes In sessionOwnership table

### PR DESCRIPTION
**Pull request by VISHNUDAS-tunerlabs**
BUG : https://katha.shikshalokam.org/bug-view-1233.html
Major changes : 
1. type [pk] added in sessionOwnership table
2. mentor_id column name changes to user_id
3. changes in functions
4. Migration added

### CODE REVIEW

File: src/database/models/sessionOwnerships.js [hunk 1]
```diff
@@ -17,6 +17,14 @@ module.exports = (sequelize, DataTypes) => {
 				allowNull: false,
 				primaryKey: true,
 			},
+			type: {
+				type: DataTypes.STRING,
+				allowNull: false,
+				primaryKey: true,
+				validate: {
+					isIn: [['MENTOR', 'CREATOR']],
```
Comment (joffinjoy) [inline src/database/models/sessionOwnerships.js:25]: Please post the validation error that happens when there is a validation and the response the server would give.

File: src/database/queries/sessions.js [hunk 2]
```diff
@@ -748,11 +758,12 @@ exports.deactivateAndReturnMentorSessions = async (userId) => {
 		const foundSessionOwnerships = await SessionOwnership.findAll({
 			attributes: ['session_id'],
 			where: {
-				mentor_id: userId,
+				user_id: userId,
 			},
 			raw: true,
 		})
-		const sessionIds = foundSessionOwnerships.map((ownership) => ownership.session_id)
+		const sessionIdsSet = new Set(foundSessionOwnerships.map((ownership) => ownership.session_id))
```
Comment (joffinjoy) [inline src/database/queries/sessions.js:765]: Move this logic the query function itself and then handle it with the flag as we discussed. Let me know if there are any issues with that logic.

File: src/database/models/sessionOwnerships.js (hunk 1 above)
Comment (VISHNUDAS-tunerlabs) [reply src/database/models/sessionOwnerships.js:25]: @joffinjoy ValidationError [SequelizeValidationError]: Validation error: Validation isIn on type failed

Comment (VISHNUDAS-tunerlabs) [reply src/database/models/sessionOwnerships.js:25]: @joffinjoy added specific error handling in sessionOwnership creation function.

File: src/database/queries/sessionOwnership.js [hunk 3]
```diff
@@ -1,8 +1,49 @@
...
+	}
+}
+
+/**
+ * Find session ownership data based on provided filter and options.
+ *
+ * @param {Object} filter - The filter criteria for the query.
+ * @param {Object} options - Additional options for the query.
+ * @param {boolean} returnUniqueSessionIds - Whether to return unique session IDs.
+ * @returns {Promise<Array>|Promise<Array<number>>|Error} - Returns an array of session ownership data
```
Comment (joffinjoy) [inline src/database/queries/sessionOwnership.js:27]: Correct this.

**Pull request merged**